kernel/src/engine/default/parquet.rs

```diff
@@ -27,6 +34,66 @@ pub struct DefaultParquetHandler<E: TaskExecutor> {
     readahead: usize,
 }
 
+/// Metadata of a parquet file, currently just includes the file metadata but will expand to
+/// include file statistics and other metadata in the future.
+#[derive(Debug)]
+pub struct ParquetWriteMetadata {
+    file_meta: FileMeta,
+}
+
+impl ParquetWriteMetadata {
+    pub fn new(file_meta: FileMeta) -> Self {
+        Self { file_meta }
+    }
+
+    // convert ParquetMetadata into a record batch which matches the 'write_metadata' schema
+    fn as_record_batch(
```

Comment:
`write_parquet_file` is called from `write_parquet` [here](https://github.com/delta-incubator/delta-kernel-rs/pull/393/files#diff-2e773d8269410d5fbb084046aba4c56bee5e5cf75df83eafdd0b03aa0454cf9aR117)

I think this design is a consequence of `write_parquet` only taking one file at a time, and that kernel _has_ to go into `EngineData` at that point. If you could pass multiple files to `write_parquet` then you'd write them all and build one record batch with all the metadata. But since we do it one at a time, _and_ kernel has to "save" each metadata so it can loop over it in the commit, we have to do this one row approach.

What we really want here (i think) is just an "opaque" type that allows being turned into `EngineData`, potentially in a batch. 

So `ParquetMetadata` would become a type at the kernel level, not just in the engine, but it's just a marker. And then kernel saves up all the `ParquetMetadata`s (whatever the engine wants to return for them) and then calls `into_engine_data(metadatas)` (another new kernel level api), where the engine can turn all the metadata at once into its `EngineData`.

Given the scope of that, if we like it, maybe we stick with this and tackle that as a follow-up?

Reply:
Yea and for context on the parquet file writing - I think we likely do want to support writing multiple files at once - java kernel does something similar taking an Iterator of column batches to write. I've aimed to derisk this whole thing by implementing this just within our default engine (no write parquet file engine API) and we can expand the API later?

Reply:
and yea @nicklan I haven't thought about that sort of trait-based API (that would be a trait that needs `into_engine_data(..)` right?) that sounds good and yea probably try to do simple here and i'll make an issue to follow up?

Reply:
I'm okay with that. Note that the trait probably wouldn't have `into_engine_data`, but rather _something_ in the engine would need an `into_engine_data(Vec<TheNewTrait>)` (or maybe iter). You need a collection of them all at once

Reply:
just opened #466 since I'm on a roll and love making new issues